presto-main/src/test/java/com/facebook/presto/operator/TestGenericPageProcessor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.SequencePageBuilder;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.PageBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.operator.FilterFunctions.TRUE_FUNCTION;
+import static com.facebook.presto.operator.ProjectionFunctions.singleColumn;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.testing.TestingConnectorSession.SESSION;
+import static org.testng.Assert.assertEquals;
+
+public class TestGenericPageProcessor
+{
+    private static final int POSITIONS = 100;
+    private final List<Type> types = ImmutableList.of(BIGINT, VARCHAR);
+    private final PageProcessor processor = new GenericPageProcessor(TRUE_FUNCTION, ImmutableList.of(singleColumn(types.get(0), 0), singleColumn(types.get(1), 1)));
+    private final Page page = createPage(types, false);
+    private final PageBuilder pageBuilder = new PageBuilder(types);
+
+    @Test
+    public void testProcess()
+            throws Exception
+    {
+        processor.process(SESSION, page, 0, page.getPositionCount(), pageBuilder);
+        Page outputPage = pageBuilder.build();
+        assertEquals(outputPage.getPositionCount(), page.getPositionCount());
```

Comment:
Can we use one of the test assertion methods to verify the contents of the output page?